tests/bootstrap_project/test/project_test.dart

```diff
@@ -0,0 +1,432 @@
+import 'package:path/path.dart' as path;
+
+void main() async {
+  Directory.current = path.join(Directory.current.path, '..', '..');
+  final rootPath = Directory.current.path;
+  final cliPath = path.join(rootPath, 'tools', 'serverpod_cli');
+  final tempPath = path.join(rootPath, 'temp');
+
+  final timeout = Timeout(Duration(minutes: 5));
+
+  setUpAll(() async {
+    await Process.run(
+      'dart',
+      ['pub', 'global', 'activate', '-s', 'path', '.'],
+      workingDirectory: cliPath,
+    );
+
+    await Process.run('mkdir', ['temp'], workingDirectory: rootPath);
+    Directory.current = tempPath;
+  });
+
+  group('Given a clean state', () {
+    final (projectName, commandRoot) = createRandomProjectName(tempPath);
+
+    late Process createProcess;
+
+    tearDown(() async {
+      createProcess.kill();
+
+      await Process.run(
+        'docker',
+        ['compose', 'down', '-v'],
+        workingDirectory: commandRoot,
+      );
+    });
+
+    test(
+        'when creating a new project then the project is created successfully and can be booted',
+        () async {
+      createProcess = await Process.start(
+        'serverpod',
+        ['create', projectName, '-v'],
+        workingDirectory: tempPath,
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      createProcess.stdout.transform(Utf8Decoder()).listen(print);
+      createProcess.stderr.transform(Utf8Decoder()).listen(print);
+
+      var createProjectExitCode = await createProcess.exitCode;
+      expect(createProjectExitCode, 0);
+
+      final docker = await Process.start(
+        'docker',
+        ['compose', 'up', '--build', '--detach'],
+        workingDirectory: commandRoot,
+      );
+
+      docker.stdout.transform(Utf8Decoder()).listen(print);
+      docker.stderr.transform(Utf8Decoder()).listen(print);
+
+      var dockerExitCode = await docker.exitCode;
+
+      expect(
+        dockerExitCode,
+        0,
+        reason: 'Docker with postgres failed to start.',
+      );
+
+      var startProcess = await Process.start(
+        'dart',
+        ['bin/main.dart', '--role', 'maintenance'],
+        workingDirectory: commandRoot,
+      );
+
+      startProcess.stdout.transform(Utf8Decoder()).listen(print);
+      startProcess.stderr.transform(Utf8Decoder()).listen(print);
+
+      var startProjectExitCode = await startProcess.exitCode;
+      expect(startProjectExitCode, 0);
+    });
+  }, timeout: timeout);
+  group('Given a clean state', () {
+    var (projectName, commandRootPath) = createRandomProjectName(tempPath);
+    final (serverDir, flutterDir, clientDir) =
+        createProjectFolderPaths(projectName);
+
+    tearDownAll(() async {
+      await Process.run(
+        'docker',
+        ['compose', 'down', '-v'],
+        workingDirectory: commandRootPath,
+      );
+    });
+
+    group('when creating a new project', () {
+      setUpAll(() async {
+        await Process.run(
```

Comment:
Suggestion - Assert the exit code is != 0 in the setup.

This will help us identify when the pre-requisite for the test is not fulfilled.